### PR TITLE
Migrate remaining v5 docs to lib-v1

### PR DESCRIPTION
**Pull request by cshfang**
#### Description of changes:
This PR migrates the v5 JS docs for API (REST), Geo, Predictions and Client Configuration to lib-v1

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [x] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/lib-v1/restapi/cancel/index.mdx [hunk 1]
```diff
@@ -4,7 +4,11 @@ import { INTEGRATION_FILTER_OPTIONS } from '@/utils/filter-data.ts';
 
 <ChooseFilterPage
   directoryPath="/ChooseFilterPage"
-  address="/lib/predictions/intro"
+<<<<<<<< HEAD:src/pages/lib-v1/storage/custom-plugin/index.mdx
+  address="/lib-v1/storage/custom-plugin"
+========
+  address="/lib-v1/restapi/cancel"
+>>>>>>>> d9922c288 (Update storage (#6115)):src/pages/lib-v1/restapi/cancel/index.mdx
```
Comment (AllanZhengYP) [inline src/pages/lib-v1/restapi/cancel/index.mdx:11]: Resolve merge conflict.

File: src/pages/lib-v1/restapi/authz/q/platform/[platform].mdx [hunk 2]
```diff
@@ -45,3 +45,11 @@ import android2 from '/src/fragments/lib-v1/restapi/native_common/authz/common.m
...
 
 <Fragments fragments={{ flutter: flutter3 }} />
+
+import js0 from '/src/fragments/lib-v1/restapi/js/authz.mdx';
+
+<Fragments fragments={{ js: js0 }} />
+
+import reactnative0 from '/src/fragments/lib-v1/restapi/js/authz.mdx';
+
+<Fragments fragments={{ 'react-native': reactnative0 }} />
```
Comment (AllanZhengYP) [inline src/pages/lib-v1/restapi/authz/q/platform/[platform].mdx:55]: Super nit: 
```suggestion
import js4 from '/src/fragments/lib-v1/restapi/js/authz.mdx';

<Fragments fragments={{ js: js4 }} />

import reactnative5 from '/src/fragments/lib-v1/restapi/js/authz.mdx';

<Fragments fragments={{ 'react-native': reactnative5 }} />
```

File: src/pages/lib-v1/restapi/fetch/q/platform/[platform].mdx [hunk 3]
```diff
@@ -45,3 +45,11 @@ import android2 from '/src/fragments/lib-v1/restapi/android/fetch.mdx';
...
 
 <Fragments fragments={{ flutter: flutter3 }} />
+
+import js0 from '/src/fragments/lib-v1/restapi/js/fetch.mdx';
+
+<Fragments fragments={{ js: js0 }} />
+
+import reactnative0 from '/src/fragments/lib-v1/restapi/js/fetch.mdx';
+
+<Fragments fragments={{ 'react-native': reactnative0 }} />
```
Comment (AllanZhengYP) [inline src/pages/lib-v1/restapi/fetch/q/platform/[platform].mdx:55]:
```suggestion

import js4 from '/src/fragments/lib-v1/restapi/js/fetch.mdx';

<Fragments fragments={{ js: js4 }} />

import reactnative5 from '/src/fragments/lib-v1/restapi/js/fetch.mdx';

<Fragments fragments={{ 'react-native': reactnative5 }} />
```

File: src/pages/lib-v1/restapi/getting-started/q/platform/[platform].mdx [hunk 4]
```diff
@@ -44,3 +44,11 @@ import android2 from '/src/fragments/lib-v1/restapi/native_common/getting-starte
...
 
 <Fragments fragments={{ flutter: flutter3 }} />
+
+import js0 from '/src/fragments/lib-v1/restapi/js/getting-started.mdx';
+
+<Fragments fragments={{ js: js0 }} />
+
+import reactnative0 from '/src/fragments/lib-v1/restapi/js/getting-started.mdx';
+
+<Fragments fragments={{ 'react-native': reactnative0 }} />
```
Comment (AllanZhengYP) [inline src/pages/lib-v1/restapi/getting-started/q/platform/[platform].mdx:54]:
```suggestion

import js4 from '/src/fragments/lib-v1/restapi/js/getting-started.mdx';

<Fragments fragments={{ js: js4 }} />

import reactnative5 from '/src/fragments/lib-v1/restapi/js/getting-started.mdx';

<Fragments fragments={{ 'react-native': reactnative5 }} />
```

File: src/pages/lib-v1/restapi/update/q/platform/[platform].mdx [hunk 5]
```diff
@@ -45,3 +45,11 @@ import android2 from '/src/fragments/lib-v1/restapi/android/update.mdx';
...
 
 <Fragments fragments={{ flutter: flutter3 }} />
+
+import js0 from '/src/fragments/lib-v1/restapi/js/update.mdx';
+
+<Fragments fragments={{ js: js0 }} />
+
+import reactnative0 from '/src/fragments/lib-v1/restapi/js/update.mdx';
+
+<Fragments fragments={{ 'react-native': reactnative0 }} />
```
Comment (AllanZhengYP) [inline src/pages/lib-v1/restapi/update/q/platform/[platform].mdx:55]:
```suggestion

import js4 from '/src/fragments/lib-v1/restapi/js/getting-started.mdx';

<Fragments fragments={{ js: js4 }} />

import reactnative5 from '/src/fragments/lib-v1/restapi/js/getting-started.mdx';

<Fragments fragments={{ 'react-native': reactnative5 }} />
```

**Review by AllanZhengYP — COMMENTED** [review]
Post some nit comments for REST-API section.

File: src/pages/lib-v1/geo/escapehatch/q/platform/[platform].mdx [hunk 6]
```diff
@@ -37,3 +37,7 @@ import ios_maintenance from '/src/fragments/lib-v1/ios-maintenance.mdx';
 import ios2 from '/src/fragments/lib-v1/geo/ios/escapehatch.mdx';
 
 <Fragments fragments={{ ios: ios2 }} />
+
+import js0 from '/src/fragments/lib-v1/geo/js/escapehatch.mdx';
+
+<Fragments fragments={{ js: js0 }} />
```
Comment (ashwinkumar6) [inline src/pages/lib-v1/geo/escapehatch/q/platform/[platform].mdx:43]:
```suggestion

import js3 from '/src/fragments/lib-v1/geo/js/escapehatch.mdx';

<Fragments fragments={{ js: js3 }} />
```

File: src/pages/lib-v1/geo/existing-resources/q/platform/[platform].mdx [hunk 7]
```diff
@@ -37,3 +37,7 @@ import ios_maintenance from '/src/fragments/lib-v1/ios-maintenance.mdx';
 import ios2 from '/src/fragments/lib-v1/geo/existing-resources.mdx';
 
 <Fragments fragments={{ ios: ios2 }} />
+
+import js0 from '/src/fragments/lib-v1/geo/existing-resources.mdx';
+
+<Fragments fragments={{ js: js0 }} />
```
Comment (ashwinkumar6) [inline src/pages/lib-v1/geo/existing-resources/q/platform/[platform].mdx:43]:
```suggestion

import js3 from '/src/fragments/lib-v1/geo/existing-resources.mdx';

<Fragments fragments={{ js: js3 }} />
```

File: src/pages/lib-v1/geo/getting-started/q/platform/[platform].mdx [hunk 8]
```diff
@@ -37,3 +37,7 @@ import ios_maintenance from '/src/fragments/lib-v1/ios-maintenance.mdx';
 import ios2 from '/src/fragments/lib-v1/geo/native_common/getting_started/common.mdx';
 
 <Fragments fragments={{ ios: ios2 }} />
+
+import js0 from '/src/fragments/lib-v1/geo/js/getting-started.mdx';
+
+<Fragments fragments={{ js: js0 }} />
```
Comment (ashwinkumar6) [inline src/pages/lib-v1/geo/getting-started/q/platform/[platform].mdx:43]:
```suggestion

import js3 from '/src/fragments/lib-v1/geo/js/getting-started.mdx';

<Fragments fragments={{ js: js3 }} />
```

File: src/pages/lib-v1/geo/maps/q/platform/[platform].mdx [hunk 9]
```diff
@@ -37,3 +37,7 @@ import ios_maintenance from '/src/fragments/lib-v1/ios-maintenance.mdx';
 import ios2 from '/src/fragments/lib-v1/geo/native_common/maps/common.mdx';
 
 <Fragments fragments={{ ios: ios2 }} />
+
+import js0 from '/src/fragments/lib-v1/geo/js/maps.mdx';
+
+<Fragments fragments={{ js: js0 }} />
```
Comment (ashwinkumar6) [inline src/pages/lib-v1/geo/maps/q/platform/[platform].mdx:43]:
```suggestion

import js3 from '/src/fragments/lib-v1/geo/js/maps.mdx';

<Fragments fragments={{ js: js3 }} />
```

File: src/pages/lib-v1/geo/search/q/platform/[platform].mdx [hunk 10]
```diff
@@ -37,3 +37,7 @@ import ios_maintenance from '/src/fragments/lib-v1/ios-maintenance.mdx';
 import ios2 from '/src/fragments/lib-v1/geo/native_common/search/common.mdx';
 
 <Fragments fragments={{ ios: ios2 }} />
+
+import js0 from '/src/fragments/lib-v1/geo/js/search.mdx';
+
+<Fragments fragments={{ js: js0 }} />
```
Comment (ashwinkumar6) [inline src/pages/lib-v1/geo/search/q/platform/[platform].mdx:43]:
```suggestion

import js3 from '/src/fragments/lib-v1/geo/js/search.mdx';

<Fragments fragments={{ js: js3 }} />
```

**Review by ashwinkumar6 — APPROVED** [review]
lgtm, Thanks chris
(nit suggestions)

**Review by AllanZhengYP — APPROVED** [review]
REST API looks good to me!

**Review by ashwinkumar6 — APPROVED** [review]

**Review by abdallahshaban557 — APPROVED** [review]
LGTM

File: src/fragments/lib/predictions/js/identify-entity.mdx
Comment (Samaritan1011001) [inline src/fragments/lib/predictions/js/identify-entity.mdx:1]: Do these predictions files also need to be moved to lib-v1?

File: src/directory/directory.mjs [hunk 11]
```diff
@@ -1475,7 +1465,7 @@ export const directory = {
           {
             title: 'Transcribe audio to text',
             route: '/lib-v1/predictions/transcribe',
-            filters: ['android', 'ios']
```
Comment (Samaritan1011001) [inline src/directory/directory.mjs:1478]: Why is android taken out here?

**Review by Samaritan1011001 — APPROVED** [review]

File: src/fragments/lib/predictions/js/identify-entity.mdx
Comment (cshfang) [reply src/fragments/lib/predictions/js/identify-entity.mdx:1]: Thanks for calling this out. These shouldn't be removed actually - Should be in https://github.com/aws-amplify/docs/pull/6128/files

File: src/directory/directory.mjs (hunk 11 above)
Comment (cshfang) [reply src/directory/directory.mjs:1478]: Not sure why Android was here in the first place. It's only supported on iOS

Comment (erinleigh90) [reply src/directory/directory.mjs:1478]: I'll double-check my predictions PR when this gets merged in to make sure android is removed there as well.

**Review by erinleigh90 — COMMENTED** [review]
For Predictions changes this looks good.

**Pull request merged**